pkg/strategy/support/strategy.go

```diff
@@ -77,14 +78,68 @@ func (stop *PercentageTargetStop) GenerateOrders(market types.Market, pos *types
 	return targetOrders
 }
 
-// ResistanceStop is a kind of stop order by detecting resistance
-type ResistanceStop struct {
-	Interval      types.Interval   `json:"interval"`
-	sensitivity   fixedpoint.Value `json:"sensitivity"`
-	MinVolume     fixedpoint.Value `json:"minVolume"`
-	TakerBuyRatio fixedpoint.Value `json:"takerBuyRatio"`
+type TrailingStopTarget struct {
+	TrailingStopCallBackRatio float64 `json:"trailingStopCallBackRatio"`
+	MinimumProfitPercentage   float64 `json:"minimumProfitPercentage"`
+}
+
+type TrailingStopControl struct {
+	symbol           string
+	market           types.Market
+	marginSideEffect types.MarginOrderSideEffectType
+
+	trailingStopCallBackRatio float64
+	minimumProfitPercentage   float64
+
+	CurrentHighestPrice fixedpoint.Value
+	OrderID             uint64
+}
+
+func NewTrailingStopControl(symbol string, market types.Market, marginSideEffect types.MarginOrderSideEffectType, trailingStopCallBackRatio float64, minimumProfitPercentage float64) *TrailingStopControl {
+	return &TrailingStopControl{
+		symbol:                    symbol,
+		market:                    market,
+		marginSideEffect:          marginSideEffect,
+		CurrentHighestPrice:       fixedpoint.NewFromInt(0),
+		trailingStopCallBackRatio: trailingStopCallBackRatio,
+		minimumProfitPercentage:   minimumProfitPercentage,
+	}
```

Comment:
you don't need NewTrailingStopControl since the constructor is simple enough, you can expand TrailingStopControl directly in the caller.